c++/src/OrcHdfsFile.cc

```diff
@@ -121,39 +116,24 @@ namespace orc {
     }
 
     uint64_t getNaturalReadSize() const override {
-      return BUF_SIZE;
+      return 1048576; //1 MB
     }
 
     void read(void* buf,
               uint64_t length,
               uint64_t offset) override {
 
-      ssize_t total_bytes_read = 0;
       size_t last_bytes_read = 0;
 
       if (!buf) {
         throw ParseError("Buffer is null");
       }
 
       hdfs::Status status;
-      char input_buffer[BUF_SIZE];
-      do{
-        //Reading file chunks
-        status = file->PositionRead(input_buffer, sizeof(input_buffer), offset, &last_bytes_read);
-        if(status.ok()) {
-          //Writing file chunks to buf
-          if(total_bytes_read + last_bytes_read >= length){
-            memcpy((char*) buf + total_bytes_read, input_buffer, length - total_bytes_read);
-            break;
-          } else {
-            memcpy((char*) buf + total_bytes_read, input_buffer, last_bytes_read);
-            total_bytes_read += last_bytes_read;
-            offset += last_bytes_read;
-          }
-        } else {
-          throw ParseError("Error reading the file: " + status.ToString());
-        }
-      } while (true);
+      status = file->PositionRead(buf, static_cast<size_t>(length), static_cast<off_t>(offset), &last_bytes_read);
```

Comment:
you still need to put this in a loop in case last_bytes_read is less than length.